core/shared/src/main/scala/matryoshka/package.scala

```diff
@@ -561,16 +561,33 @@ package object matryoshka {
     *
     * @group algtrans
     */
-  def attributeAlgebraM[F[_]: Functor, M[_]: Functor, A](f: AlgebraM[M, F, A]):
-      AlgebraM[M, F, Cofree[F, A]] =
-    fa => f(fa ∘ (_.head)) ∘ (Cofree(_, fa))
+  object attributeAlgebraM {
+    def apply[T] = new PartiallyApplied[T]
+
+    final class PartiallyApplied[T] {
+      def apply[F[_]: Functor, M[_]: Functor, A]
+        (f: AlgebraM[M, F, A])
+        (implicit T: Birecursive.Aux[T, EnvT[A, F, ?]])
+          : AlgebraM[M, F, T] =
+        fa => f(fa ∘ (_.project.ask)) ∘ (a => EnvT((a, fa)).embed)
```

Comment:
I haven’t thought through if this is actually possible, but could this be
```scala
(implicit T: Recursive.Aux[T, EnvT[A, F, ?]])
  : TransformM[M, F, EnvT[A, F, ?], T]
```
and just remove the `.embed`? I’m not sure it really gains us anything (`transCataM` will need the `Corecursive` instance anyway), but I try to keep the constraints down.